src/supernova/mod.rs

```diff
@@ -239,7 +240,8 @@ where
       num_circuits,
     );
     let mut cs: ShapeCS<G2> = ShapeCS::new();
-    let _ = circuit_secondary.synthesize(&mut cs);
+    let res = circuit_secondary.synthesize(&mut cs);
+    assert!(res.is_ok(), "circuit synthesis failed");
```

Comment:
aka aka `circuit_secondary.synthesize(&mut cs).expect("circuit synthesis failed");`

Reply:
Thanks! I thought I was missing something. Looks nicer like this.